clients/src/main/java/org/apache/kafka/clients/consumer/internals/ListOffsetsRequestManager.java

```
@@ -109,38 +109,42 @@ public NetworkClientDelegate.PollResult poll(final long currentTimeMs) {
+     * timestamp).The future will complete when the requests responses are received and
+     * processed, following a call to {@link #poll(long)}
      */
-    public CompletableFuture<Map<TopicPartition, Long>> fetchOffsets(final Set<TopicPartition> partitions,
-                                                                     final long timestamp,
-                                                                     final boolean requireTimestamps) {
-        metadata.addTransientTopics(offsetFetcherUtils.topicsForPartitions(partitions));
+    public CompletableFuture<Map<TopicPartition, OffsetAndTimestamp>> fetchOffsets(
+            final Map<TopicPartition, Long> timestampsToSearch,
+            final boolean requireTimestamps) {
+        if (timestampsToSearch.isEmpty()) {
+            return CompletableFuture.completedFuture(Collections.emptyMap());
+        }
+        metadata.addTransientTopics(offsetFetcherUtils.topicsForPartitions(timestampsToSearch.keySet()));
 
-        Map<TopicPartition, Long> timestampsToSearch = partitions.stream()
-                .collect(Collectors.toMap(Function.identity(), tp -> timestamp));
         ListOffsetsRequestState listOffsetsRequestState = new ListOffsetsRequestState(
                 timestampsToSearch,
                 requireTimestamps,
                 offsetFetcherUtils,
                 isolationLevel);
         listOffsetsRequestState.globalResult.whenComplete((result, error) -> {
             metadata.clearTransientTopics();
-            log.debug("Fetch offsets completed for partitions {} and timestamp {}. Result {}, " +
-                    "error", partitions, timestamp, result, error);
+            log.debug("Fetch offsets completed for partitions and timestamps {}. Result {}, " +
+                    "error", timestampsToSearch, result, error);
+        }).exceptionally(error -> {
+            log.error(error.getMessage());
+            return null;
```

Are we sure that the callers check for `null`?

---

Definitely not, and this would be swallowing the exception which is not the intention here (This was only added for experimenting with the completableFuture exception handling but then forgot to remove it, good catch)